scripts/microcleanfs.py

```diff
@@ -29,22 +32,30 @@
 from ampy.pyboard import Pyboard, PyboardError
 from docopt import docopt
 
-
 def main(args: List[str]) -> None:
     opts = docopt(__doc__, argv=args)
     port = opts['PORT']
+    force = opts['--force']
 
     print('Connecting to {}'.format(port), file=sys.stderr)
     board = Pyboard(port)
     files = Files(board)
+    exclude_files = ['boot.py']     # Specifying subdirectories DOES NOT work as they will
+                                    # be deleted when the parent directory is deleted.
+                                    # Specifying top level directories DOES work.
 
     print('Removing the contents of the file system')
     wait_for_board()
     for name in files.ls():
-        try:
-            files.rm(name)
-        except (RuntimeError, PyboardError):
-            files.rmdir(name)
+        if force or not any(name == x for x in exclude_files):
```

Comment:
A more readable version of this check is:

```python
if force or name not in exclude_files: ...
```

Reply:
I could swear I tried that but it may have been late and I was making things difficult for myself :-)  I'll figure out why I created that monstrosity and update appropriately.